.pre-commit-config.yaml

```
@@ -0,0 +1,17 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v5.0.0
+  hooks:
+  - id: trailing-whitespace
+    exclude: ^testing/Baseline
+  - id: end-of-file-fixer
+    exclude: ^testing/Baseline|examples/.*Baseline.*
+
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  rev: v0.8.1
+  hooks:
+    - id: ruff-format
+    - id: ruff
+      args: [--fix]
```

Nit: inconsistent formatting.

```suggestion
  - id: ruff-format
  - id: ruff
    args: [--fix]
```